src/renderers/webgl/WebGLGeometries.js

```diff
@@ -6,6 +6,20 @@ function WebGLGeometries( gl, attributes, info, bindingStates ) {
 
 	const geometries = new WeakMap();
 	const wireframeAttributes = new WeakMap();
+	const useWeakRef = typeof WeakRef !== 'undefined' && typeof FinalizationRegistry !== 'undefined';
+	let finalizer = null;
+	let weakRefGeometries = null;
+
+	if ( useWeakRef ) {
+
+		weakRefGeometries = new Set();
+		finalizer = new FinalizationRegistry( handle => {
```

Comment:
Um, it seems `FinalizationRegistry` is currently not supported in IE and Safari.